### ISSUE TITLE
Issue while using "drop_remainder=True" on validation set with TPU

### ISSUE THREAD

**Issue by dimitreOliveira** [tensorflow/tensorflow#36932]
**System information** 
- Have I written custom code (as opposed to using a stock
example script provided in TensorFlow): Yes
- Tensorflow version: `2.1.0`

About hardware and software system information I'm using Kaggle kernels so more information can be found here: https://github.com/Kaggle/docker-python

**Describe the current behavior**
I'm getting errors like the ones below, when I use the argument `drop_remainder=True` on the `dataset.batch()` function, but this happens only with the validation set:
![tpu erros](https://user-images.githubusercontent.com/16668746/74946458-18a92100-53d8-11ea-85ad-d9a2aaa1d0bc.PNG)

The issue was further discussed in [this thread on the Kaggle forum](https://www.kaggle.com/c/flower-classification-with-tpus/discussion/130717)

**Describe the expected behavior**
The model was supposed to train normally like when I don't use the `drop_remainder=True` argument.

**Standalone code to reproduce the issue** 
Link for the Kaggle kernel: https://www.kaggle.com/dimitreoliveira/flower-classification-with-tpus-eda-and-baseline

**Other info / logs** 
Here is a quick summary of the experiments:
- [Version 32](https://www.kaggle.com/dimitreoliveira/flower-classification-with-tpus-eda-and-baseline?scriptVersionId=28912464): I've not used `drop_remainder=True` on any set and got no error.
- [Version 33](https://www.kaggle.com/dimitreoliveira/flower-classification-with-tpus-eda-and-baseline?scriptVersionId=28930091): I've used `drop_remainder=True` on train and validation sets and got error.
- [Version 34](https://www.kaggle.com/dimitreoliveira/flower-classification-with-tpus-eda-and-baseline?scriptVersionId=28931447): I've used `drop_remainder=True` only on train and got no error.
- [Version 35](https://www.kaggle.com/dimitreoliveira/flower-classification-with-tpus-eda-and-baseline/notebook?scriptVersionId=28932681): I've used `drop_remainder=True` only on validation and got error.

So it seems the errors are related to using `drop_remainder=True` on the validation set.

**Comment by ymodak** [comment]
Is this still an issue?

**Comment by dimitreOliveira** [comment]
Hi @ymodak , this was mostly related to memory issues, so it is no longer an issue, I'm closing.
